lib/smack/SmackInstGenerator.cpp

```diff
@@ -882,4 +832,179 @@ void SmackInstGenerator::visitMemSetInst(llvm::MemSetInst& msi) {
   emit(rep->memset(msi));
 }
 
+void SmackInstGenerator::generateUnModeledCall(llvm::CallInst* ci) {
+  if (ci->getFunctionType()->getReturnType()->isVoidTy()) {
+    WARN("ignoring call to "+ci->getCalledFunction()->getName().str());
+    emit(Stmt::skip());
+  } else {
+    errs() << "warning: over-approximate call to : "
+      << ci->getCalledFunction()->getName() << "\n";
+    emit(rep->call(ci->getCalledFunction(), *ci));
+  }
+}
+
+void SmackInstGenerator::visitIntrinsicInst(llvm::IntrinsicInst& ii) {
+  processInstruction(ii);
+
+  static const auto f16UpCast = [this] (CallInst* ci) {
+    // translation: $f := $fpext.bvhalf.*($rmode, $bitcast.bv16.bvhalf($i));
+    auto argT = rep->type(ci->getArgOperand(0)->getType());
+    auto retT = rep->type(ci->getFunctionType()->getReturnType());
+    if (SmackOptions::FloatEnabled && SmackOptions::BitPrecise)
+      emit(Stmt::assign(
+            rep->expr(ci),
+            Expr::fn(indexedName("$fpext", {Naming::HALF_TYPE, retT}),
+              {Expr::id(Naming::RMODE_VAR),
+              Expr::fn(indexedName("$bitcast", {argT, Naming::HALF_TYPE}),
+                  rep->expr(ci->getArgOperand(0)))})));
+    else
+      generateUnModeledCall(ci);
+  };
+
+  static const auto f16DownCast = [this] (CallInst* ci){
+    // translation: assume($bitcast.bv16.bvhalf($i) == $fptrunc.bvfloat.bvhalf($rmode, $f));
+    auto argT = rep->type(ci->getArgOperand(0)->getType());
+    auto retT = rep->type(ci->getFunctionType()->getReturnType());
+    if (SmackOptions::FloatEnabled && SmackOptions::BitPrecise) {
+      emit(Stmt::assume(
+            Expr::eq(
+              Expr::fn(indexedName("$fptrunc", {argT, Naming::HALF_TYPE}),
+                Expr::id(Naming::RMODE_VAR),
+                rep->expr(ci->getArgOperand(0))),
+              Expr::fn(indexedName("$bitcast", {retT, Naming::HALF_TYPE}),
+                rep->expr(ci)))));
+    } else
+      generateUnModeledCall(ci);
+  };
+
+  static const auto assignBVFuncApp = [this] (CallInst* ci, std::string fnBase) {
```

Comment:
This function is not being use anywhere any more it seems. We should probably remove it.

Reply:
Done.